### PR TITLE
Fix double start with Mesos

### PR DESCRIPTION
**Pull request by jimenez**
There is no difference between create and start in Mesos so Docker run was technically doing 2 starts.
This PR is to prevent a second start to happen.
Signed-off-by: Isabel Jimenez <contact@isabeljimenez.com>

### CODE REVIEW

File: api/primary.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ var routes = map[string]map[string]handler{
 		"/containers/{name:.*}/unpause":       proxyContainerAndForceRefresh,
 		"/containers/{name:.*}/rename":        postRenameContainer,
 		"/containers/{name:.*}/restart":       proxyContainerAndForceRefresh,
-		"/containers/{name:.*}/start":         proxyContainerAndForceRefresh,
+		"/containers/{name:.*}/start":         postContainersStart,
```
Comment (abronan) [inline api/primary.go]: Does this have a side effect on the Swarm side? (if we omit the force refresh)

/cc @vieux

File: test/integration/mesos/api/logs.bats [hunk 2]
```diff
@@ -29,6 +29,7 @@ function teardown() {
 	# verify
 	run docker_swarm logs test_container
 	[ "$status" -eq 0 ]
+[ "${#lines[@]}" -eq 3 ]
```
Comment (abronan) [inline test/integration/mesos/api/logs.bats]: nit: small indentation issue :ghost:

File: cluster/mesos/cluster.go [hunk 3]
```diff
@@ -177,6 +177,15 @@ func (c *Cluster) UnregisterEventHandler(h cluster.EventHandler) {
 	c.eventHandlers.UnregisterEventHandler(h)
 }
 
+// StartContainer starts a container
+func (c *Cluster) StartContainer(container *cluster.Container) error {
+	// if the container was started less than a second ago, do not start it
+	if time.Now().Unix()-container.Created > 1 {
```
Comment (tnachen) [inline cluster/mesos/cluster.go]: Can we provide more comments why we even check this and why this is possible?

Comment (tnachen) [inline cluster/mesos/cluster.go]: And also if somehow there is a delay more than 1 second of the second request we will still see double start right?

Comment (vieux) [inline cluster/mesos/cluster.go]: yes, but if the container is already started, start does nothing so it's not an issue.

The issue is for cases like `docker run busybox ls`

File: api/primary.go (hunk 1 above)
Comment (vieux) [inline api/primary.go]: nope seems good to me

File: test/integration/mesos/api/logs.bats (hunk 2 above)
Comment (jimenez) [inline test/integration/mesos/api/logs.bats]: fixed

**Comment by vieux** [comment]
LGTM, test failure unrelated ping @abronan

**Comment by abronan** [comment]
LGTM

**Comment by nishanttotla** [comment]
@vieux @abronan @dongluochen one side-effect of this change is that the HostConfig isn't being passed on when starting a container, causing issues like #1781. I think it's a bug and needs to be fixed.

**Comment by vieux** [comment]
@nishanttotla ok let's fix it, it's quite trivial, but

HostConfig is deprecated see the note in  https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#start-a-container the hostconfig should be sent at create instead.

**Pull request merged**